src/sagemaker/jumpstart/factory/estimator.py

```diff
@@ -595,6 +596,25 @@ def _add_env_to_kwargs(
             value,
         )
 
+    environment = getattr(kwargs, "environment", {}) or {}
+    if (
+        environment.get(SAGEMAKER_GATED_MODEL_S3_URI_TRAINING_ENV_VAR_KEY)
+        and str(environment.get("accept_eula", "")).lower() != "true"
+    ):
+        model_specs = verify_model_region_and_return_specs(
+            model_id=kwargs.model_id,
+            version=kwargs.model_version,
+            region=kwargs.region,
+            scope=JumpStartScriptScope.TRAINING,
+            tolerate_deprecated_model=kwargs.tolerate_deprecated_model,
+            tolerate_vulnerable_model=kwargs.tolerate_vulnerable_model,
+            sagemaker_session=kwargs.sagemaker_session,
+        )
+        if model_specs.is_gated_model():
```

Comment:
Question: shouldn't this be the outer `if` statement here? And we would throw an error in all of the cases (the `environment` does not contain the special key, `accept_eula` is missing, `accept_eula` is false, etc.)?

Reply:
+1, this is odd, why bother retrieving the specs if you're not going to use them?

Reply:
fwiw, this won't involve another s3 call, it's just reading from memory